### PR TITLE
Feature/adding file attachments

### PR DESCRIPTION
**Pull request by s-ashwinkumar**
1. Added File attachments so that user can create a comment with attachments or add attachments to an existing comments. Also includes delete operation on a single attachment or all attachments of a comment.
2. Once a comment is modified, story.comment still returns the old comment as it returns from the instance variable, added an optional parameter 'reload' to re-fetch the comment.
3. Small refactoring of client to avoid repeating methods definition for get, put, post, delete (more meta programming can be used for that class)
4. Added specs for the features above and updated the README to show the usage of added features.

Please let me know about any corrections. I have tried to follow the same pattern as it is currently.

### CODE REVIEW

**Comment by forest** [comment]
Thanks for contributing. I've been swamped but will get to this in the next week or so.

**Comment by s-ashwinkumar** [comment]
@forest Thank you for letting me know !

File: lib/tracker_api/endpoints/attachment.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      def initialize(client)
+        @client = client
+      end
+
+      def create(comment, file)
+        data = client.post("/projects/#{comment.project_id}/uploads", body: FileUtility.get_file_upload(file)).body
+        Resources::FileAttachment.new({ comment: comment }.merge(data))
+      end
+
+      # This orphans the file, deletes source, but appears in the comments
```
Comment (forest) [inline lib/tracker_api/endpoints/attachment.rb]: Can we remove the commented out code?

You can put TODOs or notes in comments at the top of the class if you want.

File: lib/tracker_api/file_utility.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module TrackerApi
+	class FileUtility
```
Comment (forest) [inline lib/tracker_api/file_utility.rb]: Please fix this white space.

File: lib/tracker_api/resources/file_attachment.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+      # TODO : Implement download properly. 
+      # Look at Attchment#download for more details
+      # The big_url actually has the AWS S3 link for the file
+      # def download
+      #   file_data = Endpoints::Attachment.new(comment.client).download(download_url)
+      #   File.open(filename, 'wb') { |fp| fp.write(file_data) }
+      # end
+
+      def to_upload_hash
```
Comment (forest) [inline lib/tracker_api/resources/file_attachment.rb]: What is this used for?

File: lib/tracker_api/endpoints/attachment.rb (hunk 1 above)
Comment (s-ashwinkumar) [inline lib/tracker_api/endpoints/attachment.rb]: @forest Thank you. Sorry, for some weird reason I never got any notifications. Let me get to these this week.  Will convert this to TODO

File: lib/tracker_api/file_utility.rb (hunk 2 above)
Comment (s-ashwinkumar) [inline lib/tracker_api/file_utility.rb]: (y)

**Comment by s-ashwinkumar** [comment]
@forest handled the CR changes. Thank you and sorry for the delay.

**Comment by forest** [comment]
No problem. Thanks.

**Pull request merged**